### PR TITLE
chore: unify circus logs to bentoml + fix circus config parsing for api_server

### PR DESCRIPTION
**Pull request by aarnphm**
Signed-off-by: Aaron Pham <29749331+aarnphm@users.noreply.github.com><!--- Thanks for sending a pull request!

Addressing logs format not unified by the changes on `--reload`. Include a fix for `reload_delay` being parsed as wrong value.

### CODE REVIEW

File: bentoml/_internal/utils/circus.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
 from __future__ import annotations
 
-import typing
+import typing as t
 import logging
 import pathlib
+from typing import final
```
Comment (aarnphm) [inline bentoml/_internal/utils/circus.py:6]:
```suggestion
```

File: bentoml/_internal/utils/circus.py [hunk 2]
```diff
@@ -77,25 +76,25 @@ class BentoChangeReloader(CircusPlugin):
...
+        assert "bento_identifier" in config, "`bento_identifier` is required"
+        assert "working_dir" in config, "`working_dir` is required"
+
+        super().__init__(*args, **config)  # type: ignore (unfinished types for circus)
+
+        self.name = self.config.get("name")
+        working_dir: str = self.config["working_dir"]
+
+        # circus/plugins/__init__.py:282 -> converts all given configs to dict[str, str]
+        self.reload_delay: float = float(self.config.get("reload_delay", 1))
```
Comment (aarnphm) [inline bentoml/_internal/utils/circus.py:91]: @bojiang remove the conversion for float here will reproduce the bug

**Review by bojiang — APPROVED** [review]

File: bentoml/_internal/server/cli/api_server.py [hunk 3]
```diff
@@ -20,49 +22,48 @@
...
+    "--bind",
+    type=click.STRING,
+    required=True,
+    help="Bind address sent to circus. This address accepts the following values: 'tcp://127.0.0.1:3000','unix:///tmp/bento_api.sock', 'fd://12'",
+)
+@click.option(
+    "--runner-map",
+    type=click.STRING,
+    envvar="BENTOML_RUNNER_MAP",
+    help="Path to runner map file, default sets to envars `BENTOML_RUNNER_MAP`",
```
Comment (parano) [inline bentoml/_internal/server/cli/api_server.py:35]: this is not a file path, but a json str

File: bentoml/_internal/server/__init__.py [hunk 4]
```diff
@@ -123,7 +124,7 @@ def serve_development(
         watchers,
         sockets=circus_sockets,
         plugins=plugins,
-        debug=True,
```
Comment (parano) [inline bentoml/_internal/server/__init__.py:126]: Always use true, this is dev server only

File: bentoml/_internal/server/__init__.py [hunk 5]
```diff
@@ -17,6 +17,7 @@
 from ..utils import reserve_free_port
 from ..utils.uri import path_to_uri
 from ..utils.circus import create_standalone_arbiter
+from ..configuration import get_debug_mode
```
Comment (aarnphm) [inline bentoml/_internal/server/__init__.py:20]:
```suggestion

```

File: bentoml/_internal/log.py [hunk 6]
```diff
@@ -38,17 +47,13 @@ class TraceFormatter(Formatter):
     """
 
     def __init__(self):
-        Formatter.__init__(
-            self,
-            fmt=TRACED_LOG_FORMAT,
-            datefmt=DATE_FORMAT,
-        )
-        self.control_formmater = Formatter(SERVICE_LOG_FORMAT, datefmt=DATE_FORMAT)
+        Formatter.__init__(self, fmt=TRACED_LOG_FORMAT, datefmt=DATE_FORMAT)
```
Comment (parano) [inline bentoml/_internal/log.py:50]: Should this be super.__init?

Comment (aarnphm) [reply bentoml/_internal/log.py:50]: yep

Comment (aarnphm) [reply bentoml/_internal/log.py:50]: Screenshot 2022-05-28 at 02 03 42

Comment (aarnphm) [reply bentoml/_internal/log.py:50]: Screenshot 2022-05-28 at 02 04 21

File: bentoml/_internal/log.py [hunk 7]
```diff
@@ -9,25 +11,32 @@
...
-        record.component = ServiceContext.component_name
-
-        return Filter.filter(self, record)
+    def filter(self, record: logging.LogRecord) -> bool:
+        if "circus" in record.name:
+            component_name = "circus"
+        elif "asyncio" in record.name:
+            component_name = "asyncio"
+        else:
+            component_name = ServiceContext.component_name
```
Comment (ssheng) [inline bentoml/_internal/log.py:30]: Not a blocker but why do we want `circus` or `asyncio` as the component name, as opposed to using the actual component as the name?

Comment (aarnphm) [reply bentoml/_internal/log.py:30]: I was thinking to separate these logs out, better separation and helps us know where the source of the logs come from.

File: bentoml/_internal/log.py (hunk 6 above)
Comment (bojiang) [reply bentoml/_internal/log.py:50]: That's because we have `self` to be the first argument.
Just "super().__init__(fmt=xxx, datefmt=xxx)" works, I think.

**Pull request merged**